### PR TITLE
Fix for Issue #3793: add IllegalArgumentException

### PR DESCRIPTION
**Pull request by jkoehler22**

### CODE REVIEW

**Comment by jkoehler22** [comment]
#build

**Comment by jkoehler22** [comment]
Added new FAT and used it to test locally.

**Comment by jkoehler22** [comment]
From the PB: The failed testcases are not related to my change.

**Comment by jkoehler22** [comment]
#build

File: dev/com.ibm.ws.jaxrs.2.1.common/src/com/ibm/ws/jaxrs21/sse/LibertySseEventSinkImpl.java [hunk 1]
```diff
@@ -121,6 +122,9 @@ public boolean isClosed() {
                     response.getOutputStream().flush();
 
                     return CompletableFuture.completedFuture(eventContents);
+                } catch (InternalServerErrorException ex) {
+                    handleException(ex, future, event);
+                    throw new IllegalArgumentException("No suitable message body writer for OutboundSseEvent created with data " + event.getData() + " and mediaType " + event.getMediaType() + ". The data contained within the OutboundSseEvent must match the mediaType.");
```
Comment (andymc12) [inline dev/com.ibm.ws.jaxrs.2.1.common/src/com/ibm/ws/jaxrs21/sse/LibertySseEventSinkImpl.java]: My main concern here is that there are different scenarios that might also cause an `InternalServerErrorException` other than no suitable MBW.  Is it possible to be more specific in checking the exception to ensure that the reason for the exception is due to no suitable MBW?

**Comment by jkoehler22** [comment]
#build

**Comment by jkoehler22** [comment]
build issues  :(

#build

**Comment by jkoehler22** [comment]
#build

**Comment by jkoehler22** [comment]
The errors in the PB are not related to this change.

**Pull request merged**